proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -123,12 +123,12 @@ public boolean shouldHandle(PacketWrapper packet) throws Exception
     private enum State
     {
 
-        HANDSHAKE, STATUS, PING, USERNAME, ENCRYPT, FINISHING, CONFIGURING;
+        HANDSHAKE, STATUS, PING, USERNAME, ENCRYPT, FINISHING, CONFIGURING, EVENT_LOGIN, EVENT_PING;
     }
 
     private boolean canSendKickMessage()
     {
-        return thisState == State.USERNAME || thisState == State.ENCRYPT || thisState == State.FINISHING || thisState == State.CONFIGURING;
+        return thisState == State.USERNAME || thisState == State.ENCRYPT || thisState == State.FINISHING || thisState == State.CONFIGURING || thisState == State.EVENT_LOGIN;
```

Comment:
maybe we can reorder event_ping before username in the enum and check ordinal >= USERNAME.ordinal or so?

Reply:
i think it should be fine now